src/librustc/hir/mod.rs

```diff
@@ -934,7 +934,8 @@ pub struct BodyId {
 #[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
 pub struct Body {
     pub arguments: HirVec<Arg>,
-    pub value: Expr
+    pub value: Expr,
+    pub is_generator: bool,
```

Comment:
I've pushed a commit that just canonicalizes on a boolean, I think `IsGenerator` was just a holdover from previous days.